edweiss-app/components/lectures/remotecontrol/abstractRmtCtl.tsx

```diff
@@ -50,19 +51,65 @@ interface AbstractRmtCrlProps {
 export const AbstractRmtCrl: React.FC<AbstractRmtCrlProps & LightDarkProps> = ({ handleRight, handleLeft, handleMic, isRecording, lang, setLang, curPageProvided, totPageProvided, courseNameString, lectureIdString }) => {
 
     // Modal References
+    const modalRefTimer = useRef<BottomSheetModal>(null);
     const modalRefLangSelect = useRef<BottomSheetModal>(null);
 
+
+    const [timer, setTimer] = useState(0);
+    const [recall, setRecall] = useState(0);
+    const [isRunning, setIsRunning] = useState(false);
+    const [isCritical, setIsCritical] = useState(timer < recall);
+    const [timerColor, setTimerColor] = useState<'red' | 'green'>(isCritical ? 'red' : 'green');
+
+
+    useEffect(() => {
+        if (timer >= 0) {
+            let interval: NodeJS.Timeout | null = null;
+            if (isRunning) {
+                interval = setInterval(() => { if (timer > 0) setTimer(prevTimer => prevTimer - 1); }, 1000);
+            } else if (!isRunning && timer !== 0) { clearInterval(interval!); }
+            return () => clearInterval(interval!);
+        }
+    }, [isRunning, timer]);
+
+    useEffect(() => {
+        if (isRunning && timer === recall) {
+            Vibration.vibrate([50, 300, 50, 300, 50, 300]);
+            setIsCritical(true);
+        }
+
+    }, [isRunning, timer])
+
+    useEffect(() => {
+        if (isCritical) {
+            setTimerColor('red');
+        } else {
+            setTimerColor('green');
+        }
+    }, [isCritical])
+
+
+    const formatTime = (seconds: number) => {
+        const getSeconds = `0${seconds % 60}`.slice(-2);
+        const minutes = Math.floor(seconds / 60);
+        const getMinutes = `0${minutes % 60}`.slice(-2);
+        const getHours = `0${Math.floor(seconds / 3600)}`.slice(-1);
+        return `${getHours}:${getMinutes}:${getSeconds}`;
+    };
+
     return (
         <>
             <RouteHeader disabled title={"Lecture's Slides"} />
 
             <TView borderColor='text' m={'md'} mt={'xl'} mb={'xl'} b={0.5} backgroundColor='base' radius={'lg'}>
 
                 {/* STRC text, set language and timer */}
-                <TView mt={17} mb={50} justifyContent='center' alignItems='center'>
+                <TView mt={'sm'} mb={'lg'} justifyContent='center' alignItems='center'>
```

Comment:
Good thing to use strings to define the size instead of number. This is good fore understanding